internal/raft/message/convert.go

```diff
@@ -0,0 +1,971 @@
+	msgListJoin := &List_Join{&Command_Join{}}
+	msgListJoin.Join.Natural = cmd.Natural
+	msgListJoin.Join.Type = ConvertCommandToMessageJoinType(cmd.Type)
+	msgListJoin.Join.Filter = ConvertCommandToMessageExpr(cmd.Filter)
+	msgListJoin.Join.Left = ConvertCommandToMessageList(cmd.Left)
+	msgListJoin.Join.Right = ConvertCommandToMessageList(cmd.Right)
+	return msgListJoin
+}
+
+// ConvertCommandToMessageListLimit converts a command.Limit to a message.List_Limit
+func ConvertCommandToMessageListLimit(cmd *command.Limit) *List_Limit {
+	msgListLimit := &List_Limit{&Command_Limit{}}
+	msgListLimit.Limit.Limit = ConvertCommandToMessageExpr(cmd.Limit)
+	msgListLimit.Limit.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListLimit
+}
+
+// ConvertCommandToMessageListOffset converts a command.Offset to a message.List_Offset
+func ConvertCommandToMessageListOffset(cmd *command.Offset) *List_Offset {
+	msgListOffset := &List_Offset{&Command_Offset{}}
+	msgListOffset.Offset.Offset = ConvertCommandToMessageExpr(cmd.Offset)
+	msgListOffset.Offset.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListOffset
+}
+
+// ConvertCommandToMessageListDistinct converts a command.Distinct to a message.List_Distinct
+func ConvertCommandToMessageListDistinct(cmd *command.Distinct) *List_Distinct {
+	msgListDistinct := &List_Distinct{&Command_Distinct{}}
+	msgListDistinct.Distinct.Input = ConvertCommandToMessageList(cmd.Input)
+	return msgListDistinct
+}
+
+// ConvertCommandToMessageRepeatedExprSlice converts a [][]command.Expr to a [][]message.Expr
+func ConvertCommandToMessageRepeatedExprSlice(cmd [][]command.Expr) []*RepeatedExpr {
+	msgRepeatedExprSlice := []*RepeatedExpr{}
+	for i := range cmd {
+		msgRepeatedExpr := &RepeatedExpr{}
+		for j := range cmd[i] {
+			msgRepeatedExpr.Expr = append(msgRepeatedExpr.Expr, ConvertCommandToMessageExpr(cmd[i][j]))
+		}
+		msgRepeatedExprSlice = append(msgRepeatedExprSlice, msgRepeatedExpr)
+	}
+	return msgRepeatedExprSlice
+}
+
+// ConvertCommandToMessageListValues converts a command.Values to a message.List_Values
+func ConvertCommandToMessageListValues(cmd *command.Values) *List_Values {
+	msgListValues := &List_Values{}
+	msgListValues.Values.Expr = ConvertCommandToMessageRepeatedExprSlice(cmd.Values)
+	return msgListValues
+}
+
+// ConvertCommandToMessageList converts
+func ConvertCommandToMessageList(cmd command.List) *List {
+	msgList := &List{}
+	switch cmd := cmd.(type) {
+	case *command.Scan:
+		msgList.List = ConvertCommandToMessageListScan(cmd)
+	case *command.Select:
+		msgList.List = ConvertCommandToMessageListSelect(cmd)
+	case *command.Project:
+		msgList.List = ConvertCommandToMessageListProject(cmd)
+	case *command.Join:
+		msgList.List = ConvertCommandToMessageListJoin(cmd)
+	case *command.Limit:
+		msgList.List = ConvertCommandToMessageListLimit(cmd)
+	case *command.Offset:
+		msgList.List = ConvertCommandToMessageListOffset(cmd)
+	case *command.Distinct:
```

Comment:
Friendly reminder of the pointers, please check if we actually use pointers in the compiler output